src/EventStore.Core/Services/Storage/StorageWriterService.cs

```diff
@@ -447,10 +447,15 @@ public class StorageWriterService<TStreamId> : IHandle<SystemMessage.SystemInit>
 				} else {
 					// SOFT DELETE
 					var metastreamId = _systemStreams.MetaStreamOf(streamId);
-					var expectedVersion = _indexWriter.GetStreamLastEventNumber(metastreamId);
-
-					if (_indexWriter.GetStreamLastEventNumber(streamId) < 0 && expectedVersion < 0) {
-						var result = new CommitCheckResult<TStreamId>(CommitDecision.WrongExpectedVersion, streamId,
+					var metadataLastEventNumber = _indexWriter.GetStreamLastEventNumber(metastreamId);
```

Comment:
@shaan1337 Thanks for pointing out the breaking change. Please let me know what you think of my alternate solution : 

- Instead of messing with projection subsystem code to handle this "special" case (one solution is, as you pointed : check currentversion), what if I intercept reply to client and transform `StreamDeletedException`  to `WrongExpectedVersionException`; something along the lines of following 

```var httpEnvelope = new SendToHttpEnvelope(_networkSendQueue, manager, Format.DeleteStreamCompleted, ConfigureResponse);
var envelope = new CallbackEnvelope(message => {
	if (message is ClientMessage.DeleteStreamCompleted completed && completed.Result == OperationResult.StreamDeleted) {
		httpEnvelope.ReplyWith(new ClientMessage.DeleteStreamCompleted(completed.CorrelationId, OperationResult.WrongExpectedVersion, "Stream is already deleted.", completed.CurrentVersion, completed.PreparePosition, completed.CommitPosition));
	} else {
		httpEnvelope.ReplyWith(message);
	}
});```

Reply:
Making changes to projection subsytem to handle "special" case and my solution, both, to me, seems a little "hacky".
But my solution has advantage that projection code remains clean and logical.
Please let me know.

Reply:
> but i think the reason was to keep things simple and to avoid having to update all clients with a new error code. can you please confirm?

yes i think that's right. it would be weird if soft deleting a stream that doesn't exist actually _created_ a stream (the metadatastream) so we reject that append, and i think wrongexpectedversion is the only existing error code that made some sense - but i agree that it is confusing in the ExpectedVersion.Any case

Reply:
As per @shaan1337 's suggestion, I have made changes to projection subsystem instead, to maintain contract with TCP/GRPC/Web clients